### PR TITLE
pulse_connect_secure: add syslog priority format parsing support

### LINKED ISSUE #2546: Pulse Secure Connect integration "Provided Grok expressions do not match field value"
I installed experimental integration and got grok parser errors.
Below are examples of logs where:

z.z.z.z - pulse secure connect ip address
username - name of the user
REALM - name of the realm
ROLE - name of the role (can be empty)
x.x.x.x - ip address of connected host
pcs-name - hostname
HC_POLICY - host checker policy name
XX-XX-XX-XX-XX-XX - client host mac address

Provided Grok expressions do not match field value: [<174>1 2022-01-19T10:19:35+01:00 z.z.z.z PulseSecure: - - - 2022-01-19 10:19:35 - pcs-name - [x.x.x.x] username(REALM)[ROLE] - Primary authentication successful for username/REALM from x.x.x.x]

Provided Grok expressions do not match field value: [<174>1 2022-01-19T10:23:16+01:00 z.z.z.z PulseSecure: - - - 2022-01-19 10:23:16 - pcs-name - [x.x.x.x] username(REALM)[ROLE] - Host Checker policy 'HC_POLICY' passed on host 'x.x.x.x' address 'XX-XX-XX-XX-XX-XX' for user 'username'.]

Provided Grok expressions do not match field value: [<174>1 2022-01-19T10:24:04+01:00 z.z.z.z PulseSecure: - - - 2022-01-19 10:24:04 - pcs-name - [x.x.x.x] username(REALM)[ROLE] - Syslog server y.y.y.y (facility LOCAL5, filter Standard, type UDP, interface Global) removed from Events logs]

Provided Grok expressions do not match field value: [<174>1 2022-01-19T10:23:06+01:00 z.z.z.z PulseSecure: - - - 2022-01-19 10:23:06 - pcs-name - [127.0.0.1] System()[] - The current virus signature list imported successfully.]

Provided Grok expressions do not match field value: [<174>1 2022-01-19T10:23:06+01:00 z.z.z.z PulseSecure: - - - 2022-01-19 10:23:06 - pcs-name - [127.0.0.1] System()[] - The current virus signature list downloaded successfully from 'https://download.pulsesecure.net/software/av/uac/epupdate_hist.xml']

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This adds support for syslog priority value parsing.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

First commit is mechanical; this just runs the new version of elastic-package on to reduce diff noise in commit two.

## How to test this PR locally

`elastic-package test` in package/pulse_connect_secure

## Related issues

- Closes #2546.

## Screenshots

N/A

### CODE REVIEW

File: packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ processors:
       field: event.original
       patterns:
         - '%{SYSLOGTIMESTAMP} %{SYSLOGHOST:host.hostname} %{INT} %{TIMESTAMP_ISO8601:_tmp.timestamp} %{IP:observer.ip} PulseSecure: - - - %{DATE2} - %{SYSLOGHOST:observer.name} - \[%{IPORHOST:client.address}\] %{USERNAME:user.name}?\(%{DATA:pulse_secure.realm}?\)\[%{DATA:pulse_secure.role}\] - %{GREEDYDATA:message}'
+        - '^<%{NONNEGINT}>%{NUMBER}? %{TIMESTAMP_ISO8601:_tmp.timestamp} %{IP:observer.ip} PulseSecure: - - - %{DATE2} - %{SYSLOGHOST:observer.name} - \[%{IPORHOST:client.address}\] %{USERNAME:user.name}?\(%{DATA:pulse_secure.realm}?\)\[%{DATA:pulse_secure.role}\] - %{GREEDYDATA:message}'
```
Comment (legoguy1000) [inline packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:26]: I would combine the 2 patterns and just add the `<%{NONNEGINT}>%{NUMBER}?` as a optional item. Besides that, the pattern seems the same thought u changed the `host.*` to `observer.*` which might have been the right thing to begin with.

Comment (efd6) [reply packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:26]: Thanks. Yes, it is identical beyond the prefix (no change to `host.*`. The issue with just making the PRI optional is that the `%{SYSLOGTIMESTAMP} %{SYSLOGHOST:host.hostname} %{INT} ` prefix components do not appear in test cases that prompted the change, so they need to be made optional too, and then an order for the case that they both appear has to be decided. What would you suggest there?

Comment (legoguy1000) [reply packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:26]: Didn't spot that difference, THen the best thing i'd do is consolidate the similar sections into sub grok patterns as much as you can so its easier to maintain long term

Comment (efd6) [reply packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: PTAL

File: packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ processors:
   - grok:
       field: event.original
       patterns:
-        - '%{SYSLOGTIMESTAMP} %{SYSLOGHOST:host.hostname} %{INT} %{TIMESTAMP_ISO8601:_tmp.timestamp} %{IP:observer.ip} PulseSecure: - - - %{DATE2} - %{SYSLOGHOST:observer.name} - \[%{IPORHOST:client.address}\] %{USERNAME:user.name}?\(%{DATA:pulse_secure.realm}?\)\[%{DATA:pulse_secure.role}\] - %{GREEDYDATA:message}'
+        - '^(<%{NONNEGINT}>%{NUMBER}?|%{SYSLOGTIMESTAMP} %{SYSLOGHOST:host.hostname} %{INT}) %{TIMESTAMP_ISO8601:_tmp.timestamp} %{IP:observer.ip} PulseSecure: - - - %{DATE2} - %{SYSLOGHOST:observer.name} - \[%{IPORHOST:client.address}\] %{USERNAME:user.name}?\(%{DATA:pulse_secure.realm}?\)\[%{DATA:pulse_secure.role}\] - %{GREEDYDATA:message}'
```
Comment (andrewkroh) [inline packages/pulse_connect_secure/data_stream/log/elasticsearch/ingest_pipeline/default.yml:25]: How about `NONNEGINT:log.syslog.priority:long` to store the value?

https://www.elastic.co/guide/en/ecs/current/ecs-log.html#field-log-syslog-priority

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**